zc.recipe.egg_/src/zc/recipe/egg/egg.py

```diff
@@ -57,40 +57,34 @@ def working_set(self, extra=()):
         This is intended for reuse by similar recipes.
         """
         options = self.options
-        b_options = self.buildout['buildout']
+        buildout_section = self.buildout['buildout']
 
         # Backward compat. :(
         options['executable'] = sys.executable
 
-        distributions = [
-            r.strip()
-            for r in options.get('eggs', self.name).split('\n')
-            if r.strip()]
-        orig_distributions = distributions[:]
-        distributions.extend(extra)
-
-        if self.buildout['buildout'].get('offline') == 'true':
-            ws = zc.buildout.easy_install.working_set(
-                distributions,
-                [options['develop-eggs-directory'], options['eggs-directory']]
-                )
-        else:
-            ws = zc.buildout.easy_install.install(
-                distributions, options['eggs-directory'],
-                links=self.links,
-                index=self.index,
-                path=[options['develop-eggs-directory']],
-                newest=self.buildout['buildout'].get('newest') == 'true',
-                allow_hosts=self.allow_hosts)
-
-        return orig_distributions, ws
+        return working_set(
+            distributions=[
+                r.strip()
+                for r in options.get('eggs', self.name).split('\n')
+                if r.strip()
+            ],
+            develop_eggs_dir=options['develop-eggs-directory'],
+            eggs_dir=options['eggs-directory'],
+            offline=(buildout_section.get('offline') == 'true'),
+            newest=(buildout_section.get('newest') == 'true'),
+            links=self.links,
+            index=self.index,
+            allow_hosts=self.allow_hosts,
+            extra=extra,
+            )
 
     def install(self):
         reqs, ws = self.working_set()
         return ()
 
     update = install
 
+
```

Comment:
I agree with the suggestions. But how can I do that? My git-fu is weak. Perhaps close this PR and do a new one?